commands/newmember.ts

```diff
@@ -0,0 +1,78 @@
+import { exception } from "console";
+import { Collection, GuildMember, Message, Role, RoleResolvable } from "discord.js";
+import { Command, CommandDefinition, Action } from ".";
+
+export const description: CommandDefinition = {
+    name: "Update New Member Role",
+    description: "Scans through the server's users and updates the New Member role. Adds the role to anybody with no class role, and removes it from people that have added a class.",
+    usage: ["!newmember"],
+    keys: ["newmember"],
+};
+
+var staffList = ["Admin", "Mod", "ESS", "CSSA", "IEEE", "Server Developer", "Server Booster", "Club Representative", "Bot", "Omnipotent"]
+
+function noperms(exception: any, message: Message) {
+    message.channel.send(
+      `I don't have permissions to set someone's role.`
+    );
+  }
+
+export const action: Action = (message: Message, key: string) => {
+
+    //Check if the caller is a Mod or Admin
+    if(message.member?.roles.cache.some((role => role.name === 'Mod')) || message.member?.roles.cache.some(role => role.name === 'Admin')) {
+        message.reply("Beginning scan...");
+
+        var newMemberRole: Role | null = null;
+
+        //Get role for "New Member"
+        message.guild?.roles.cache.forEach(role => {
```

Comment:
To eliminate this null-check, check if `message.guild` exists earlier to guarantee the command is being called in a server.